### PR TITLE
applications: nrf_desktop: Fix config channel

### PR DESCRIPTION
**Pull request by MarekPieta**
Change fixes config channel and enables DFU support for keyboard (it was missing for DebugMCUBoot configuration).

### CODE REVIEW

File: scripts/hid_configurator/configurator.py [hunk 1]
```diff
@@ -75,16 +75,16 @@ def __repr__(self):
...
         if data_field_len < 0:
             logging.error('Response too short')
             return None
 
         # Report ID is not included in the feature report from device
-        fmt = '<HBBB{}s'.format(data_field_len)
+        fmt = '<BHBBB{}s'.format(data_field_len)
 
-        (rcpt, event_id, status, data_len, data) = struct.unpack(fmt, response_raw)
+        (report_type, rcpt, event_id, status, data_len, data) = struct.unpack(fmt, response_raw)
```
Comment (pdunaj) [inline scripts/hid_configurator/configurator.py]: report_id

Comment (MarekPieta) [reply scripts/hid_configurator/configurator.py]: done

File: scripts/hid_configurator/configurator.py [hunk 2]
```diff
@@ -75,16 +75,16 @@ def __repr__(self):
...
         if data_field_len < 0:
             logging.error('Response too short')
             return None
 
         # Report ID is not included in the feature report from device
-        fmt = '<HBBB{}s'.format(data_field_len)
+        fmt = '<BHBBB{}s'.format(data_field_len)
 
-        (rcpt, event_id, status, data_len, data) = struct.unpack(fmt, response_raw)
+        (report_id, rcpt, event_id, status, data_len, data) = struct.unpack(fmt, response_raw)
```
Comment (pdunaj) [inline scripts/hid_configurator/configurator.py]: we also need to check if report_id is what we wait for

Comment (MarekPieta) [reply scripts/hid_configurator/configurator.py]: Seems so, especially if we use USB (appllication may set first byte to any chosen value)

**Pull request merged**